specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotSecuritySolutionAnalytics.json

```diff
@@ -0,0 +1,916 @@
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "IoTSeverityMetrics": {
+      "type": "object",
+      "description": "Severity metrics",
+      "properties": {
+        "high": {
+          "type": "integer",
+          "description": "count of high severity items"
+        },
+        "medium": {
+          "type": "integer",
+          "description": "count of medium severity items"
+        },
+        "low": {
+          "type": "integer",
+          "description": "count of low severity items"
+        }
+      }
+    },
+    "IoTSecuritySolutionAnalyticsModel": {
+      "type": "object",
+      "description": "Security Analytics of a security solution",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Security Solution  Aggregated Alert data",
+          "$ref": "#/definitions/IoTSecuritySolutionAnalyticsModelProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
+        }
+      ]
+    },
+    "IoTSecuritySolutionAnalyticsModelProperties": {
+      "description": "Security Analytics of a security solution properties",
+      "properties": {
+        "metrics": {
+          "type": "object",
+          "$ref": "#/definitions/IoTSeverityMetrics",
+          "description": "Security Analytics of a security solution",
+          "readOnly": true
+        },
+        "unhealthyDeviceCount": {
+          "type": "integer",
+          "readOnly": true,
+          "description": "number of unhealthy devices"
+        },
+        "devicesMetrics": {
+          "description": "The list of devices metrics by the aggregated date.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "properties": {
+              "date": {
+                "type": "string",
+                "format": "date-time",
+                "description": "the date of the metrics"
+              },
+              "devicesMetrics": {
+                "type": "object",
+                "$ref": "#/definitions/IoTSeverityMetrics",
+                "description": "devices alerts count by severity."
+              }
+            }
+          }
+        },
+        "topAlertedDevices": {
+          "description": "The list of top 3 devices with the most attacked.",
+          "type": "object",
+          "$ref": "#/definitions/IoTSecurityAlertedDevicesList"
+        },
+        "mostPrevalentDeviceAlerts": {
+          "description": "The list of most prevalent 3 alerts.",
+          "type": "object",
+          "$ref": "#/definitions/IoTSecurityDeviceAlertsList"
+        },
+        "mostPrevalentDeviceRecommendations": {
+          "description": "The list of most prevalent 3 recommendations.",
+          "type": "object",
+          "$ref": "#/definitions/IoTSecurityDeviceRecommendationsList"
+        }
```

Comment:
All of these properties can be modeled as arrays. The pattern with `value` array and `nextLink` is only intended for collection GETs. How would you even pass in multiple individual skip tokens to page this effectively. You should model as an array. In fact, it seems like your examples are doing that already.